### PR TITLE
Lint getStates. No changes in logic

### PR DESCRIPTION
**Pull request by jonganc**
I just linted the code using eslint. While the code looks good, I did not yet have time to test it out. 
@wtee  Let me know if you don't like any of the eslint options and we can discuss changing them.
While linting, I find one probable bug, marked CHECK.

I changed some eslint options to allow for patterns that both Wes and I seem to find unobjectionable.

### CODE REVIEW

**Comment by nausicaa-rose** [comment]
The part Jon marked to check is correct, or at least intended. In the hypothetical case this script were run without an initial file of groups, that would trigger the 'ENOENT' error since there's no file to read from. Setting groupsJSON to null means the getCurrentGroups will return null, which means when updateGroups checks for the existence of currentGroups if current groups is null the function will get the state info for fetchedGroups and skip comparing the groups from the Indivisible website to groups we already have information for. 

In short if that line of code returns null, getStates.js creates groups.json with the groups from the Indivisible website.

I don't know if that's the best approach or not and will be happy to make changes, but that's why it's the way it is.

**Comment by jonganc** [comment]
I think that makes sense and work well. 

Also, I now realize that the original code also worked fine. I think I got confused by the value being returned by the promise. In any case, I think the current and original behavior is the same anyway. It's up to you if you want to tweak the way the code is written, however.

**Comment by jonganc** [comment]
@alecfrancesconi Can we merge this?

**Comment by alecfrancesconi** [comment]
Yes.
Alec

**Comment by jonganc** [comment]
? Only you have write privileges.
